include/aos/common/tools/string.hpp

```diff
@@ -473,6 +472,7 @@ class String : public Array<char> {
         return {cDigits[high], cDigits[low]};
     }
 
+private:
```

Comment:
You have byte array to hex and back. Why do you need this function public? It looks strange, especially with  Pair<char, char>. I can imagine how to use it if it returns char[2] for example. But pair of chars it is really some very specific case.

Reply:
it is required for percent encoded params in url. 
0x2025u => "%20%25"